modules/dynamic-plugins/con-preinstalled-dynamic-plugins.adoc

```diff
@@ -2,9 +2,26 @@
 
 = Preinstalled dynamic plugins
 
-{product} is preinstalled with a selection of dynamic plugins. The dynamic plugins that require custom configuration are disabled by default.
-
-For a complete list of dynamic plugins that are preinstalled in this release of {product-short}, see the xref:rhdh-supported-plugins[Dynamic plugins support matrix].
+{product} is preinstalled with a selection of dynamic plugins. For a complete list of dynamic plugins that are preinstalled in this release of {product-short}, see the xref:rhdh-supported-plugins[Dynamic plugins support matrix].
+
+The following preinstalled dynamic plugins are enabled by default:
```

Comment:
this list will be reduced in 1.3 - see https://issues.redhat.com/browse/RHIDP-3187

Reply:
also this is probably better managed by code generation so we don't have to manually manage this. If you're ever thinking "should I hard code a list of things" please ask yourself first "or can we generate this from the plugins themselves?"

Reply:
@nickboldt Fair enough but there'll be only be 5 enabled if we remove the the tech preview plugins, so it's not currently a huge overhead to manage either way.

If we are modifying the script could we look at generating the three tables as separate reference modules which would be more in keeping with modular documentation guidelines.

BTW I'm all for automating the generation of these tables for consistency and accuracy.

Reply:
How does having them in 3 files vs. 1 make anything easier?

Reply:
Having discussed this with @Preeticp, while we see great value in using the script to generate reliable table data, we'd suggest that each table should be a separate reference module and just contain the table data e.g. `ref-rh-supported-plugins.adoc`, `ref-rh-tech-preview-plugins.adoc` and `ref-community-plugins.adoc`.

The text content that is currently generated by the script would then be contained in an assembly (or concept module) that would include the reference modules. This approach would be in keeping with docs modularization guidelines and enable docs to update this text content at any time without the need to run the script or re-generate the table data.

Can the script be modified to use this approach? If not, or if it involves a significant amount of re-factoring, we can review further.

Reply:
So your thought here is that instead of a single template with section titles and notes/warnings, you would have 6 files for the three table titles and the table content all chunked separately? 

That is doable - though it would require some significant reworking of the code, and ought to be handled in a separate JIRA as it's out of scope for what we were trying to achieve with the current one. 

If I'm going to spend time on anything related to making the table **content** better, perhaps my time would be best spent on **adding content** (the missing description data), rather than on refactoring/reorganizing the existing content that an end user won't ever see or know about? 

6 files vs. 1 doesn't make the act of contributing content any simpler, but to your point, yes, you would not have to regen the table content (which aside takes about 30 seconds and could be added to the build.sh script so it runs on all pushes &/or pull requests). 

Can you explain it to me like I'm 5 years old, how this change will benefit customers or streamline your migration to AEM?

Reply:
@nickboldt From a Docs point-of-view, we need the script to produce the following three files:

- ref-rh-supported-plugins.adoc
- ref-rh-tech-preview-plugins.adoc
- ref-community-plugins.adoc

The files should contain just the tabular data from the three tables that we have agreed to include in the restructured plugins documentation.

All other text content will be included in an assembly, concept module (or snippet) which will be created and maintained by the Docs team.

c.c. @Preeticp